revision to utils-html to use [[]] instead of $

mhines-usgs opened the pull request:
and a new test for format comments still in the works. will add two or three more small tests for some of these functions but not go crazy since the long term goal is to move most of the html functions to java.

mhines-usgs committed:
new test for utils-html and some rework
will continue to add tests

mhines-usgs committed:
Merge branch 'master' of github.com:USGS-R/repgen

thongsav-usgs commented on tests/testthat/test-utils-html.R:
```diff
@@ -0,0 +1,49 @@
+
+test_that('does it replace the escaped characters with real html breaks?', {
+  library(jsonlite)
+  readings <- fromJSON('
```
FYI, as you have it written, the json is actually a "reportObject" and not a "readings" object because you have the top level nesting in the json. If you want just the json for a readings object you can drop the `{ "readings":` from the start and parse just the array. Then it would be a readings (list) object being parsed.

mhines-usgs replied:
changes to reportObject

thongsav-usgs commented on tests/testthat/test-utils-html.R:
```diff
@@ -0,0 +1,49 @@
+                       }
+                        ] }
+                       ')
+  refComm <- repgen:::getComments(readings[["readings"]][["recorderComments"]])
```
Related to comment above, this should actually be written as `reportObject[["readings"]][["recorderComments"]]`.

mhines-usgs replied:
Yeah, I figured. I'm still working on finishing this test up and will push up the corrections. Does it matter which way I go, with the reportObject or with readings?

mhines-usgs replied:
Going to go with reportObject as it's more 'real world'

mhines-usgs committed:
- finishing a test for formatComments
- revising the object to be reportObject instead of readings which was wrong
- correcting a weird html thing in formatComments i cant believe worked in the first place

mhines-usgs commented on R/utils-html.R:
```diff
@@ -92,8 +92,8 @@ printReportFeature <- function(feature, isTable=FALSE, m=NULL, mar_values=c(8, 3
   if(isEmptyOrBlank(split_comments)){return(split_comments)}
-  htmlbreaks_inside <- lapply(split_comments, gsub, pattern="\r\n", replacement="</br>")
-  htmlbreaks_end <- lapply(htmlbreaks_inside, paste0, "</br>", collapse="")
+  htmlbreaks_inside <- lapply(split_comments, gsub, pattern="\r\n", replacement="<br/>")
```
according to this http://www.w3schools.com/code/tryit.asp?filename=FBNJO5J8MUTJ

this ```</br>``` works, but it could only be working because the browser is allowing it to.
